wallet/src/backend.rs

```diff
@@ -538,6 +534,33 @@ impl<'a, Meta: Serialize + DeserializeOwned + Send> CapeWalletBackend<'a>
         Ok(address.map(Erc20Code::from))
     }
 
+    async fn wait_for_wrapped_erc20_code(
+        &mut self,
+        asset: &AssetDefinition,
```

Comment:
We should change this to take an `AssetCode`, makes it easier to call

Reply:
For the mock backend, `wait_for_wrapped_erc20_code` calls `get_wrapped_erc20_code` which takes an `AssetDefinition` to check if it's in the `erc20_registrar` table. If we're going to modify `wait_for_wrapped_erc20_code` to take an `AssetCode`, then we also need to change `erc20_registrar` and some other places (e.g., the `wrapped_erc20_registrar` field of `CapeContract`). I guess those structs/data take an `AssetDefinition` rather than `AssetCode` for some reason, so I'm not sure if we should change the type.

Reply:
We can add the `AssetCode` -> `Erc20Code` mapping to the mock EQS in `mocks.rs` and add the `AssetSponsored` event in `model.rs` and basically simulate exactly what the real EQS does. But TBH it's probably fine to do a linear search for an `AssetDefinition` with the given code for now, in the mock backend. Something like `erc20_registrar.iter().find(|definition| definition.code == code)`